File: image-train-filters-app/src/main/java/org/jesperancinha/itf/android/file/manager/FileManagerItem.java
package org.jesperancinha.itf.android.file.manager;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.io.File;
import java.io.Serializable;

@Getter
@Builder
@AllArgsConstructor
public class FileManagerItem implements Comparable<FileManagerItem>, Serializable {

    private final String filename;
    private final String date;
    private final FileType fileType;
    private final File file;

    @Override
    public int compareTo(FileManagerItem another) {
        return another.getFilename().compareTo(this.getFilename());
    }
}

File: app-g2/app/src/main/java/org/jesperancinha/itf/app/manager/FileManagerItem.kt
package org.jesperancinha.itf.app.manager

import java.io.File
import java.io.Serializable

class FileManagerItem (
    val filename: String,
    val date: String,
    val fileType: FileType,
    val file: File,
) : Comparable<FileManagerItem>, Serializable
{
    override fun compareTo(other: FileManagerItem): Int =
        this.filename.compareTo(other.filename)

}